src/menus/DecompilerContextMenu.cpp

```diff
@@ -163,22 +175,77 @@ void DecompilerContextMenu::aboutToShowSlot()
     QString progCounterName = Core()->getRegisterName("PC").toUpper();
     actionSetPC.setText(tr("Set %1 here").arg(progCounterName));
 
-    if (!annotationHere) { // To be considered as invalid
+    if (!annotationHere
+            || annotationHere->type ==
+            R_CODE_ANNOTATION_TYPE_CONSTANT_VARIABLE) { // To be considered as invalid
         actionRenameThingHere.setVisible(false);
+        copySeparator->setVisible(false);
+    } else {
+        copySeparator->setVisible(true);
+        if (annotationHere->type == R_CODE_ANNOTATION_TYPE_FUNCTION_NAME) {
+            actionRenameThingHere.setVisible(true);
+            actionRenameThingHere.setText(tr("Rename function %1").arg(QString(
+                                                                           annotationHere->reference.name)));
+        }
+        if (annotationHere->type == R_CODE_ANNOTATION_TYPE_GLOBAL_VARIABLE) {
+            RFlagItem *flagDetails = r_flag_get_i(Core()->core()->flags, annotationHere->reference.offset);
+            if (flagDetails) {
+                actionRenameThingHere.setText(tr("Rename %1").arg(QString(flagDetails->name)));
+                actionDeleteName.setText(tr("Remove %1").arg(QString(flagDetails->name)));
+                actionDeleteName.setVisible(true);
+            } else {
+                actionRenameThingHere.setText(tr("Add name"));
+            }
+        }
+    }
+    actionCopyInstructionAddress.setText(tr("Copy instruction address %1").arg(RAddressString(offset)));
```

Comment:
We want to tell them what address is going to be saved to the clipboard in a "fyi" manner
```suggestion
    actionCopyInstructionAddress.setText(tr("Copy instruction address (%1)").arg(RAddressString(offset)));
```

Reply:
Changed as suggested.